src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffendersResourceTransferImpTest.kt

```diff
@@ -1343,6 +1363,690 @@ class OffendersResourceTransferImpTest : ResourceTest() {
+            temporaryAbsenceArrival(temporaryAbsenceArrivalRequest(agencyId = "LEI"))
+              .expectBody()
+              .jsonPath("inOutStatus").isEqualTo("IN")
+              .jsonPath("status").isEqualTo("ACTIVE IN")
+              .jsonPath("agencyId").isEqualTo("LEI")
+
+            getOffender(offenderNo)
+              .isOk
+              .expectBody()
+              .jsonPath("inOutStatus").isEqualTo("IN")
+              .jsonPath("status").isEqualTo("ACTIVE IN")
+              .jsonPath("agencyId").isEqualTo("LEI")
+          }
+
+          @Test
+          internal fun `default movement reason is taken from out movement`() {
+            temporaryAbsenceArrival(temporaryAbsenceArrivalRequest(agencyId = "LEI", movementReasonCode = null))
+              .expectBody()
+              .jsonPath("inOutStatus").isEqualTo("IN")
+              .jsonPath("agencyId").isEqualTo("LEI")
+
+            getOffender(offenderNo)
+              .isOk
+              .expectBody()
+              .jsonPath("inOutStatus").isEqualTo("IN")
+              .jsonPath("agencyId").isEqualTo("LEI")
+              .jsonPath("lastMovementTypeCode").isEqualTo("TAP")
+              .jsonPath("lastMovementReasonCode").isEqualTo("C3")
+          }
+
+          @Test
+          internal fun `can override movement reason`() {
+            temporaryAbsenceArrival(temporaryAbsenceArrivalRequest(agencyId = "LEI", movementReasonCode = "C6"))
+              .expectBody()
+              .jsonPath("inOutStatus").isEqualTo("IN")
+              .jsonPath("agencyId").isEqualTo("LEI")
+
+            assertThat(dataLoaderTransaction.get { lastMovement(bookingId).movementReason?.code }).isEqualTo("C6")
+          }
+
+          @Test
+          @Disabled("BROKEN, should be fixed")
+          internal fun `when override movement reason booking status field is updated to the new reason`() {
+            temporaryAbsenceArrival(temporaryAbsenceArrivalRequest(agencyId = "LEI", movementReasonCode = "C6"))
+              .expectBody()
+              .jsonPath("inOutStatus").isEqualTo("IN")
+              .jsonPath("agencyId").isEqualTo("LEI")
+
+            getOffender(offenderNo)
+              .isOk
+              .expectBody()
+              .jsonPath("lastMovementTypeCode").isEqualTo("TAP")
+              .jsonPath("lastMovementReasonCode").isEqualTo("C6")
+          }
+
+          @Test
+          @Disabled("BROKEN, should be fixed")
+          internal fun `from address should taken from to OUT address`() {
+            temporaryAbsenceArrival(temporaryAbsenceArrivalRequest(agencyId = "LEI", movementReasonCode = null))
+              .expectBody()
+              .jsonPath("inOutStatus").isEqualTo("IN")
+              .jsonPath("agencyId").isEqualTo("LEI")
+
+            assertThat(lastMovement(bookingId)?.fromAddressId).isEqualTo(toAddressId)
+          }
+
+          @Test
+          internal fun `cell remains unchanged from when the prisoner was released on TAP`() {
```

Comment:
Is there a counter test where they get booked back into a reception cell if they released the cell on the TAP OUT?

Reply:
Oh hang on this is the bed released test! And they get received back into the same cell they released? I guess only if the bed isn't assigned to someone else?

Reply:
Yes, so oddly when they are transferred out to court or TAP and their bed is released they are placed in a virtual cell "COURT" or "TAP".

I guess it is so the prison stats still work